opm/simulators/flow/Main.hpp

```diff
@@ -0,0 +1,526 @@
+#include <flow/flow_ebos_oilwater.hpp>
+#include <flow/flow_ebos_solvent.hpp>
+#include <flow/flow_ebos_polymer.hpp>
+#include <flow/flow_ebos_foam.hpp>
+#include <flow/flow_ebos_brine.hpp>
+#include <flow/flow_ebos_energy.hpp>
+#include <flow/flow_ebos_oilwater_polymer.hpp>
+#include <flow/flow_ebos_oilwater_polymer_injectivity.hpp>
+#endif
+
+#include <opm/simulators/flow/Main.hpp>
+#include <opm/simulators/flow/FlowMainEbos.hpp>
+#include <opm/simulators/utils/moduleVersion.hpp>
+#include <opm/models/utils/propertysystem.hh>
+#include <opm/models/utils/parametersystem.hh>
+#include <opm/simulators/flow/MissingFeatures.hpp>
+
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/common/OpmLog/EclipsePRTLog.hpp>
+#include <opm/common/OpmLog/LogUtil.hpp>
+
+#include <opm/io/eclipse/rst/state.hpp>
+#include <opm/io/eclipse/ERst.hpp>
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/Parser/Parser.hpp>
+#include <opm/parser/eclipse/Parser/ParseContext.hpp>
+#include <opm/parser/eclipse/Parser/ErrorGuard.hpp>
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/checkDeck.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.hpp>
+
+#include <opm/parser/eclipse/EclipseState/Schedule/ArrayDimChecker.hpp>
+
+#if HAVE_DUNE_FEM
+#include <dune/fem/misc/mpimanager.hh>
+#else
+#include <dune/common/parallel/mpihelper.hh>
+#endif
+
+#if HAVE_MPI
+#include <opm/simulators/utils/ParallelEclipseState.hpp>
+#include <opm/simulators/utils/ParallelSerialization.hpp>
+#endif
+
+BEGIN_PROPERTIES
+
+// this is a dummy type tag that is used to setup the parameters before the actual
+// simulator.
+NEW_TYPE_TAG(FlowEarlyBird, INHERITS_FROM(EclFlowProblem));
+
+END_PROPERTIES
+
+namespace Opm
+{
+    // ----------------- Main class -----------------
+    //   For now, we will either be instantiated from main() in flow.cpp,
+    //   or from a Python pybind11 module..
+    // NOTE (March 2020): When used from a pybind11 module, we do not neccessarily
+    //   want to run the whole simulation by calling run(), it is also
+    //   useful to just run one report step at a time. According to these different
+    //   usage scenarios, we refactored the original run() in flow.cpp into this class.
+    class Main
+    {
+    private:
+        enum class FileOutputMode {
+            //! \brief No output to files.
+            OUTPUT_NONE = 0,
+            //! \brief Output only to log files, no eclipse output.
+            OUTPUT_LOG_ONLY = 1,
+            //! \brief Output to all files.
+            OUTPUT_ALL = 3
+        };
+    public:
+        Main(int argc, char** argv) : argc_(argc), argv_(argv)  {  }
+
+        int run() {
+            return main_(argc_, argv_);
+        }
+
+    private:
+
+        Opm::filesystem::path simulationCaseName_( const std::string& casename ) {
```

Comment:
Removed the `details::` namespace prefix for `simulationCaseName_()` since it appears it is no longer need?